client/app/dbaas/logs/detail/streams/alerts/streams-alerts.service.js

```diff
@@ -0,0 +1,168 @@
+            .then(operation => this._handleSuccess(serviceName, operation.data, "streams_alerts_delete_success"))
+            .catch(this.ServiceHelper.errorHandler("streams_alerts_delete_error"));
+    }
+
+    /**
+     * Get the IDs of all alerts
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @returns promise which will be resolve with a list of alert IDs
+     * @memberof StreamsAlertsService
+     */
+    getAlertIds (serviceName, streamId) {
+        return this.AlertsApiService.query({
+            serviceName,
+            streamId
+        }).$promise
+            .catch(this.ServiceHelper.errorHandler("streams_alerts_ids_loading_error"));
+    }
+
+    /**
+     * Gets the alert objects corresponding to the alertIds
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @param {any} alertIds - list of alert IDs for which alert object is to be fetched
+     * @returns promise which will be resolve with the list of alerts
+     * @memberof StreamsAlertsService
+     */
+    getAlerts (serviceName, streamId, alertIds) {
+        return this.getAlertDetails(serviceName, streamId, alertIds)
+            .catch(this.ServiceHelper.errorHandler("streams_alerts_loading_error"));
+    }
+
+    /**
+     * Gets the alert objects corresponding to the alertIds
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @param {any} alertIds - list of alert IDs for which alert object is to be fetched
+     * @returns promise which will be resolve with the list of alerts
+     * @memberof StreamsAlertsService
+     */
+    getAlertDetails (serviceName, streamId, alertIds) {
+        const promises = alertIds.map(alertId => this.getAlert(serviceName, streamId, alertId));
+        return this.$q.all(promises);
+    }
+
+    /**
+     * Gets the alert object corresponding to the alertId
+     *
+     * @param {any} serviceName
+     * @param {any} streamId
+     * @param {any} alertId - the alert ID for which alert object is to be fetched
+     * @returns promise which will be resolve with the alert
+     * @memberof StreamsAlertsService
+     */
+    getAlert (serviceName, streamId, alertId) {
+        return this.AlertsApiService.get({ serviceName, streamId, alertId }).$promise;
+    }
+
+    /**
+     * Returns a new alert object with the default properties
+     *
+     * @param {any} conditionType - the type of the condition (one of StreamsAlertsConstant.alertType)
+     * @returns the default alert object
+     * @memberof StreamsAlertsService
+     */
+    getNewAlert (conditionType) {
+        const thresholdType = conditionType === this.StreamsAlertsConstant.alertType.numeric ?
+            this.StreamsAlertsConstant.thresholdType.lower :
+            this.StreamsAlertsConstant.thresholdType.more;
+        const constraintType = this.StreamsAlertsConstant.constraintType.mean;
+        return {
```

Comment:
It's a detail but I'd have return a promise (return this.$q.when(...)) just to be consistent with the rest of the methods in the services (they all return promises)

Reply:
This is done :)